compiler/src/lnntoamm/Expr.ts

```diff
@@ -356,16 +356,23 @@ export default abstract class Expr {
             precedences[applyIdx] = dec.ref();
           }
           const applyTo = precedences[applyIdx] as Ref;
-          const fns = operators.reduce(
-            (fns, op) => [...fns, ...op.select(metadata.scope, applyTo.ty)],
-            new Array<Fn>(),
+          const [fns, retTys] = operators.reduce(
+            ([fns, retTys], op) => {
+              const [selFns, selTys] = op.select(metadata.scope, applyTo.ty);
+              fns = [...fns, ...selFns];
+              retTys = [...retTys, ...selTys];
+              return [fns, retTys];
+            },
+            [new Array<Fn>(), new Array<Type>()] as [Fn[], Type[]],
           );
+          const retTy = Type.oneOf(retTys);
```

Comment:
Which one does `Type.oneOf` choose? Just the first, or something more complex?

Reply:
when `retTy.instance()` is called, it'll return the type furthest back in the list. that's to maintain the last-defined rule